deploy/runtools/runtime_config.py

```diff
@@ -257,14 +285,53 @@ def build_sim_driver(self) -> None:
             prefix('cd sim/'):
             driverbuildcommand = f"make DESIGN={design} TARGET_CONFIG={target_config} PLATFORM_CONFIG={platform_config} PLATFORM={self.platform} {self.driver_build_target}"
             buildresult = run(driverbuildcommand)
-
-            if buildresult.failed:
-                rootLogger.info(f"{self.driver_type_message} driver build failed. Exiting. See log for details.")
-                rootLogger.info("""You can also re-run '{}' in the 'firesim/sim' directory to debug this error.""".format(driverbuildcommand))
-                sys.exit(1)
+            self.handle_failure(buildresult, 'driver build', 'firesim/sim', driverbuildcommand)
 
         self.driver_built = True
 
+    def build_sim_tarball(self, paths: List[Tuple[str, str]], tarball_name: str) -> None:
+        """ Take the simulation driver and tar it. build_sim_driver() 
+        must run before this function.  """
+        if self.tarball_built:
+            # we already built it
+            return
+
+        builddir = self.local_tarball_creation_path()
+
+        with InfoStreamLogger('stdout'):
+            cmd = f"rm -rf {builddir}"
+
```

Comment:
I'm a little wary of putting `rm -rfs` in the manager. What if builddir path get's mangled by an errant commit?

Reply:
`builddir` gets created and then all of the rsync (source, dest) pairs run and copy files into it. If files are removed from source, there is no way to catch that using rsync. The will remain at dest.  There is a flag (`--delete                 delete extraneous files from dest dirs`) but I believe this won't work because there are multiple pairs. The second pair would delete all the desired files from the first run.

If you believe files being removed from source and abandoned in dest isn't an issue, then I can remove this. If there is some other trick to delete these files in a more clean way please suggest

Reply:
I've addressed this comment by making a new folder each time. the new folder incorporates the epoc time in seconds as well as a few chars of a uuid. I've removed the `rm -rf`